src/custom-elements/src/components/spinner/spinner.tsx

```diff
@@ -0,0 +1,75 @@
+import { Component, Prop, Watch } from '@stencil/core';
+import { ICON_SIZES } from '../../constants/size';
+
+const SPINNER_COLORS = ['', 'primary', 'success', 'warning', 'danger', 'muted', 'secondary', 'light'];
+
+@Component({
+  tag: 'chi-spinner',
+  styleUrl: 'spinner.scss',
+  scoped: true
+})
+export class Spinner {
+  /**
+   *  to set size of a spinner { xs, sm, sm--2, sm--3, md, lg, xl, xxl }.
+   */
+  @Prop({ reflectToAttr: true }) size: string;
```

Comment:
@SergioQCL according to Chi if no size is specified it means it renders by default at ```sm``` size, so ```size=''``` or ```size='sm'``` are same that's why I'm not setting any default value. Do you think it would be nice if we did?

Reply:
@srinath-koneru, yes, I think it might be useful. Specially now that we are showing the documentation table within pug files and one of the columns shows the default value for a given Prop, so now we would get an undefined value.

Reply:
@SergioQCL roger I'll update that.